osf_tests/test_registration_machine_states.py

```diff
@@ -0,0 +1,147 @@
+import pytest
+import datetime
+from website import settings
+
+from osf.utils.workflows import RegistrationStates
+
+from osf.exceptions import ValidationError
+from osf_tests.factories import (
+    DraftRegistrationFactory,
+    CommentFactory
+)
+
+from osf.models import Retraction, Sanction
+
+
+from django.utils import timezone
+
+
+@pytest.mark.django_db
+class TestRegistrationMachine:
+
+    @pytest.fixture()
+    def draft_registration(self):
+        return DraftRegistrationFactory()
+
+    @pytest.fixture()
+    def comment(self):
+        return CommentFactory(content='Yo')
+
+    def test_run_submit(self, draft_registration):
+        assert draft_registration.machine_state == RegistrationStates.INITIAL.value
+        draft_registration.run_submit(draft_registration.creator)
+
+        assert not draft_registration.branched_from.is_public
+        assert draft_registration.approval.meta == {
+            'embargo_end_date': None,
+            'registration_choice': 'immediate'
+        }
+        assert draft_registration.logs.first().action == 'submitted'
+        assert draft_registration.machine_state == RegistrationStates.PENDING.value
+
+    def test_run_accept(self, draft_registration):
+        draft_registration.run_submit(draft_registration.creator)
+
+        assert draft_registration.machine_state == RegistrationStates.PENDING.value
+
+        draft_registration.run_accept(draft_registration.creator, comment='Wanna Philly Philly?')
+
+        assert draft_registration.machine_state == RegistrationStates.ACCEPTED.value
+        assert draft_registration.registered_node
+        assert draft_registration.registered_node.is_public
+
+    def test_run_accept_with_embargo(self, draft_registration):
+        draft_registration.run_submit(draft_registration.creator)
+        draft_registration.save()
+
+        end_date = timezone.now() + settings.EMBARGO_END_DATE_MIN - datetime.timedelta(days=1)
```

Comment:
Clarification: This is testing with an invalid embargo? Worth making this a separate test to be clearer?

Reply:
Yeah I'm not stressing too much about the test structure here, other then getting the basic transitions correct, that should be hammered out in the next ticket where all validation and model behaviors will be tested.